File: apple/Sources/ReactNativeWatchOS/Modifiers/MaterialActiveAppearance.swift
import SwiftUI

/// SwiftUI `.materialActiveAppearance(_:)` — watchOS 26 "Liquid Glass".
/// Forces a material/glass surface to render as active or inactive,
/// independent of whether its scene is key.
///
/// `appearance`: `automatic` (system decides), `active`, or `inactive`.
///
/// Gated to watchOS 26+ — returns the view unchanged on older OS.
enum RNWMaterialActiveAppearanceModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("materialActiveAppearance") { view, params, _ in
            guard #available(watchOS 26.0, *) else { return view }

            // `.inactive` is unavailable on watchOS; only `.active` and
            // `.automatic` exist here. "inactive" falls back to `.automatic`.
            let appearance: MaterialActiveAppearance
            switch params.string("appearance") {
            case "active":   appearance = .active
            default:         appearance = .automatic
            }
            return AnyView(view.materialActiveAppearance(appearance))
        }
    }
}

File: src/modifiers/materialActiveAppearance.ts
import { createModifier } from './createModifier';

/// Whether a material/glass surface renders in its active or inactive
/// appearance. `automatic` lets the system decide based on context.
export type MaterialActiveAppearanceMode = 'automatic' | 'active' | 'inactive';

export interface MaterialActiveAppearanceParams {
  /// The appearance mode. Defaults to `automatic`.
  appearance?: MaterialActiveAppearanceMode;
}

/// SwiftUI `.materialActiveAppearance(_:)` (watchOS 26 "Liquid Glass").
/// Forces a material to render as active or inactive regardless of whether
/// its window/scene is key. No-op on watchOS < 26.
export function materialActiveAppearance(
  appearance: MaterialActiveAppearanceMode = 'automatic'
) {
  return createModifier('materialActiveAppearance', { appearance });
}
